rpi_backlight/cli.py

```diff
@@ -97,12 +97,25 @@ def main():
         return
 
     if args.set_power:
-        if any(
-            (args.get_brightness, args.set_brightness, args.get_power, args.duration)
-        ):
-            parser.error("-p/--set-power must be used without other options")
-        backlight.power = True if args.set_power == "on" else False
+        if any((args.get_brightness, args.set_brightness, args.get_power)):
+            parser.error("-p/--set-power may only be used with -d/--duration")
+        if args.set_power == "toggle":
+            if backlight.power:
+                with backlight.fade(duration=args.duration):
+                    backlight.brightness = 0
+                backlight.power = False
+            else:
+                # Ensure brightness is 0 when we turn the display on
+                backlight.brightness = 0
+                with backlight.fade(duration=args.duration):
+                    backlight.brightness = 100
+                backlight.power = True
```

Comment:
This is not working correctly, tested using the emulator:

- Run `rpi-backlight-emulator`
- Run `rpi-backlight -p off :emulator:`
- Run `rpi-backlight -p toggle -d 1 :emulator:`

The brightness slider goes from 0 - 100% in one second, and only after that the display turns on.

Needs to be:

```suggestion
                backlight.brightness = 0
                backlight.power = True
                with backlight.fade(duration=args.duration):
                    backlight.brightness = 100
```

Reply:
After applying this patch:

```diff
diff --git a/rpi_backlight/cli.py b/rpi_backlight/cli.py
index abcd21d..c9e07fb 100644
--- a/rpi_backlight/cli.py
+++ b/rpi_backlight/cli.py
@@ -107,10 +107,9 @@ def main():
             else:
                 # Ensure brightness is 0 when we turn the display on
                 backlight.brightness = 0
+                backlight.power = True
                 with backlight.fade(duration=args.duration):
                     backlight.brightness = 100
-                backlight.power = True
-
         else:
             backlight.power = True if args.set_power == "on" else False
         return
```

![GIF](https://user-images.githubusercontent.com/19366641/78458892-c765a000-76ac-11ea-92ea-30b2a1d9358a.gif)

Reply:
@linusg 
On the Tinker Board, this change is causing the screen to turn on instantly when using 
```rpi-backlight -p toggle -d 1```

The toggling off works fine with the duration
However, the toggling on is instantaneous and ignores the duration.

The following line is turning the screen on instantly:
```
backlight.power = True
```

Because the screen turns on instantly, the following fading code does not do anything:
```
with backlight.fade(duration=args.duration):
                    backlight.brightness = 100
```

Reply:
```
with backlight.fade(duration=args.duration):
                    backlight.brightness = 0
                backlight.power = False
```

In the above code, we are setting brightness to 0 and then setting power to false. This is working for me

However, when you try and set the power to true before setting the brightness, the screen turns on immediately and ignores the duration.

Reply:
@linusg I have implemented a change. Please let me know if this is working for you now.
Works on the Tinker Board

```
if args.set_power == "toggle":
            if backlight.power:
                with backlight.fade(duration=args.duration):
                    backlight.brightness = 0
                if args.board_type == "raspberry-pi":
                    backlight.power = False
            else:
                # Ensure brightness is 0 when we turn the display on
                backlight.brightness = 0
                if args.board_type == "raspberry-pi":
                    backlight.power = True
                with backlight.fade(duration=args.duration):
                    backlight.brightness = 100
```

Reply:
Yes, I suppose that would work :+1:

Please don't use "magic values" though, so replace `"raspberry-pi"` with `BOARD_TYPES[BoardType.RASPBERRY_PI]`

Reply:
Do I replace it with
BOARD_TYPES[BoardType.RASPBERRY_PI] 
or
RASPBERRY_PI
or
BoardType.RASPBERRY_PI 

:S

Reply:
BoardType.RASPBERRY_PI works
RASPBERRY_PI is undefined
BOARD_TYPES[BoardType.RASPBERRY_PI] results in a key error